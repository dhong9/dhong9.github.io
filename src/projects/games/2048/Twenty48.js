function Twenty48_Util() {}

/**
 * Copies 2D array by value into another 2D array
 * @param {number[][]} board current game board
 * represented by 2D array of numbers
 * @returns deep copy of input board
 */
Twenty48.prototype.copyBoard = (board) => board.map((row) => [...row]);

module.exports = Twenty48_Util;
