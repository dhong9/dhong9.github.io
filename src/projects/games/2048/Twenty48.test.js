// React testing libraries
import React from "react";
import renderer from "react-test-renderer";

// Component to test
import Twenty48 from "projects/games/2048";

let realUseContext;
let useContextMock;
// Setup mock
beforeEach(() => {
  realUseContext = React.useContext;
  useContextMock = React.useContext = jest.fn();
});
// Cleanup mock
afterEach(() => {
  React.useContext = realUseContext;
});

// Define Mocks
jest.mock("components/MKBox", () => {
  const { forwardRef } = jest.requireActual("react");
  return {
    __esModule: true,
    default: forwardRef(() => <div>Mock Box</div>),
  };
});
jest.mock("layouts/sections/components/BaseLayout", () => {
  const { forwardRef } = jest.requireActual("react");
  return {
    __esModule: true,
    default: forwardRef(() => <div>Mock Layout</div>),
  };
});
jest.mock("layouts/sections/components/View", () => {
  const { forwardRef } = jest.requireActual("react");
  return {
    __esModule: true,
    default: forwardRef(() => <div>Mock View</div>),
  };
});
jest.mock("react-monaco-editor", () => {
  const { forwardRef } = jest.requireActual("react");
  return {
    __esModule: true,
    default: forwardRef(() => <div>Mock Editor</div>),
  };
});
jest.mock("draft-convert", () => {
  return {
    convertFromHTML: jest.fn(),
    convertToRaw: jest.fn(),
  };
});

describe("2048", () => {
  it("renders", () => {
    useContextMock.mockReturnValue("Test Value");
    const component = renderer.create(<Twenty48 />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
