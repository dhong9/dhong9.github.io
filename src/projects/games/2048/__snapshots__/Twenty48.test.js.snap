// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`2048 renders 1`] = `
<div>
  Mock Layout
</div>
`;
